From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Thu, 13 Apr 2017 22:42:31 +0300
Subject: net: off by one in inet6_pton()
Patch-mainline: v4.12-rc1
Git-commit: a88086e09876f468356cfec5319ff62004e36a1b
References: bsc#1095643

If "scope_len" is sizeof(scope_id) then we would put the NUL terminator
one space beyond the end of the buffer.

Fixes: b1a951fe469e ("net/utils: generic inet_pton_with_scope helper")
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 net/core/utils.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/core/utils.c
+++ b/net/core/utils.c
@@ -339,7 +339,7 @@ static int inet6_pton(struct net *net, c
 	    src + srclen != scope_delim && *scope_delim == '%') {
 		struct net_device *dev;
 		char scope_id[16];
-		size_t scope_len = min_t(size_t, sizeof(scope_id),
+		size_t scope_len = min_t(size_t, sizeof(scope_id) - 1,
 					 src + srclen - scope_delim - 1);
 
 		memcpy(scope_id, scope_delim + 1, scope_len);
